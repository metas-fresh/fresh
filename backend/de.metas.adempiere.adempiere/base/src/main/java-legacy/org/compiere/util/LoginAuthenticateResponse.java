package org.compiere.util;

import com.google.common.collect.ImmutableList;
import de.metas.security.Role;
import de.metas.user.UserId;
import lombok.Builder;
import lombok.NonNull;
import lombok.Singular;
import lombok.Value;

@Value
@Builder
public class LoginAuthenticateResponse
{
	@NonNull UserId userId;

	@Singular
	@NonNull ImmutableList<Role> availableRoles;
}
