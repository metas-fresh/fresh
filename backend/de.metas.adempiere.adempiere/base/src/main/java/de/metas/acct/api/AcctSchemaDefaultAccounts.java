package de.metas.acct.api;

<<<<<<< HEAD
=======
import de.metas.acct.Account;
>>>>>>> 3091b8e938a (externalSystems-Leich+Mehl can invoke a customizable postgREST reports (#19521))
import lombok.Builder;
import lombok.NonNull;
import lombok.Value;

/*
 * #%L
 * de.metas.adempiere.adempiere.base
 * %%
 * Copyright (C) 2018 metas GmbH
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 2 of the
 * License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this program. If not, see
 * <http://www.gnu.org/licenses/gpl-2.0.html>.
 * #L%
 */

@Value
@Builder
public class AcctSchemaDefaultAccounts
{
	@NonNull
<<<<<<< HEAD
	AccountId realizedGainAcctId;
	@NonNull
	AccountId realizedLossAcctId;

	@NonNull
	AccountId unrealizedGainAcctId;
	@NonNull
	AccountId unrealizedLossAcctId;
=======
	Account realizedGainAcct;
	@NonNull
	Account realizedLossAcct;

	@NonNull
	Account unrealizedGainAcct;
	@NonNull
	Account unrealizedLossAcct;
>>>>>>> 3091b8e938a (externalSystems-Leich+Mehl can invoke a customizable postgREST reports (#19521))
}
