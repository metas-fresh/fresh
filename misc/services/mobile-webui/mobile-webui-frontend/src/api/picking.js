import axios from 'axios';
import { apiBasePath } from '../constants';
import { unboxAxiosResponse } from '../utils';
import { useEffect, useState } from 'react';
import { QTY_REJECTED_REASON_TO_IGNORE_KEY } from '../reducers/wfProcesses';

export const usePickTargets = ({ wfProcessId }) => {
  const [loading, setLoading] = useState(false);
  const [targets, setTargets] = useState([]);
  const [tuTargets, setTuTargets] = useState([]);

  useEffect(() => {
    setLoading(true);
    getPickTargets({ wfProcessId })
      .then((data) => {
        setTargets(data.targets);
        setTuTargets(data.tuTargets);
      })
      .finally(() => setLoading(false));
  }, [wfProcessId]);

  return {
    isTargetsLoading: loading,
    targets,
    tuTargets,
  };
};

const getPickTargets = ({ wfProcessId }) => {
  return axios
    .get(`${apiBasePath}/picking/job/${wfProcessId}/target/available`)
    .then((response) => unboxAxiosResponse(response));
};

export const setPickTarget = ({ wfProcessId, target }) => {
  return axios
    .post(`${apiBasePath}/picking/job/${wfProcessId}/target`, target)
    .then((response) => unboxAxiosResponse(response));
};

export const setTUPickTarget = ({ wfProcessId, target }) => {
  return axios
    .post(`${apiBasePath}/picking/job/${wfProcessId}/target/tu`, target)
    .then((response) => unboxAxiosResponse(response));
};

export const closePickTarget = ({ wfProcessId }) => {
  return axios
    .post(`${apiBasePath}/picking/job/${wfProcessId}/target/close`)
    .then((response) => unboxAxiosResponse(response));
};

export const closeTUPickTarget = ({ wfProcessId }) => {
  return axios
    .post(`${apiBasePath}/picking/job/${wfProcessId}/target/tu/close`)
    .then((response) => unboxAxiosResponse(response));
};

export const postStepPicked = ({
  wfProcessId,
  activityId,
  lineId,
  stepId,
  huQRCode,
  qtyPicked,
  qtyRejected,
  qtyRejectedReasonCode,
  catchWeight,
  pickWholeTU,
  checkIfAlreadyPacked,
  setBestBeforeDate,
  bestBeforeDate,
  setLotNo,
  lotNo,
  isCloseTarget = false,
}) => {
  const realRejectedQtyReason =
    qtyRejectedReasonCode === QTY_REJECTED_REASON_TO_IGNORE_KEY ? null : qtyRejectedReasonCode;

  return postEvent({
    wfProcessId,
    wfActivityId: activityId,
    pickingLineId: lineId,
    pickingStepId: stepId,
    type: 'PICK',
    huQRCode,
    qtyPicked,
    qtyRejectedReasonCode: realRejectedQtyReason,
    qtyRejected,
    catchWeight,
    pickWholeTU,
    checkIfAlreadyPacked,
    setBestBeforeDate,
    bestBeforeDate,
    setLotNo,
    lotNo,
    isCloseTarget,
  });
};

export const postStepUnPicked = ({ wfProcessId, activityId, lineId, stepId, huQRCode, unpickToTargetQRCode }) => {
  return postEvent({
    wfProcessId,
    wfActivityId: activityId,
    pickingLineId: lineId,
    pickingStepId: stepId,
    type: 'UNPICK',
    huQRCode,
    unpickToTargetQRCode,
  });
};

const postEvent = (event) => {
  return axios.post(`${apiBasePath}/picking/event`, event).then((response) => unboxAxiosResponse(response));
};

export const closePickingJobLine = ({ wfProcessId, lineId }) => {
  return axios
    .post(`${apiBasePath}/picking/closeLine`, { wfProcessId, pickingLineId: lineId })
    .then((response) => unboxAxiosResponse(response));
};

export const openPickingJobLine = ({ wfProcessId, lineId }) => {
  return axios
    .post(`${apiBasePath}/picking/openLine`, { wfProcessId, pickingLineId: lineId })
    .then((response) => unboxAxiosResponse(response));
};
