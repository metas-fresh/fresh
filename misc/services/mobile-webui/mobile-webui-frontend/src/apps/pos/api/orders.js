import axios from 'axios';
import { apiBasePath } from '../../../constants';
import { toUrl, unboxAxiosResponse } from '../../../utils';
import { useEffect } from 'react';
import * as ws from '../../../utils/websocket';

export const getOpenOrders = ({ ids } = {}) => {
  const queryParams = {};
  if (ids?.length > 0) {
    queryParams.ids = ids.join(',');
  }

  return axios.get(toUrl(`${apiBasePath}/pos/orders`, queryParams)).then((response) => unboxAxiosResponse(response));
};

export const updateOrder = ({ order }) => {
  return axios.post(`${apiBasePath}/pos/orders`, order).then((response) => unboxAxiosResponse(response));
};

export const changeOrderStatusToDraft = ({ order_uuid }) => {
  return axios.post(`${apiBasePath}/pos/orders/${order_uuid}/draft`).then((response) => unboxAxiosResponse(response));
};
export const changeOrderStatusToWaitingPayment = ({ order_uuid }) => {
  return axios
    .post(`${apiBasePath}/pos/orders/${order_uuid}/waitingPayment`)
    .then((response) => unboxAxiosResponse(response));
};

export const changeOrderStatusToVoid = ({ order_uuid }) => {
  return axios.post(`${apiBasePath}/pos/orders/${order_uuid}/void`).then((response) => unboxAxiosResponse(response));
};

export const changeOrderStatusToComplete = ({ order_uuid }) => {
  return axios
    .post(`${apiBasePath}/pos/orders/${order_uuid}/complete`)
    .then((response) => unboxAxiosResponse(response));
};

export const useOrdersWebsocket = ({ terminalId, onWebsocketMessage }) => {
  useEffect(() => {
    let client;
    const topic = `/pos/orders/${terminalId}`;

    client = ws.connectAndSubscribe({
      topic,
      debug: !!window?.debug_ws,
      onWebsocketMessage: (message) => {
        onWebsocketMessage(JSON.parse(message.body));
      },
    });
    console.log(`WS connected to ${topic}`, { client });

    return () => {
      if (client) {
        ws.disconnectClient(client);
        client = null;
        console.log(`WS disconnected from ${topic}`);
      }
    };
  }, [terminalId]);
};

export const checkoutPayment = ({ order_uuid, payment_uuid }) => {
  return axios
    .post(`${apiBasePath}/pos/orders/${order_uuid}/payments/${payment_uuid}/checkout`)
    .then((response) => unboxAxiosResponse(response));
};

export const refundPayment = ({ order_uuid, payment_uuid }) => {
  return axios
    .post(`${apiBasePath}/pos/orders/${order_uuid}/payments/${payment_uuid}/refund`)
    .then((response) => unboxAxiosResponse(response));
};
