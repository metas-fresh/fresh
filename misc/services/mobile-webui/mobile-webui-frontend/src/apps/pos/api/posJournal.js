import axios from 'axios';
import { apiBasePath } from '../../../constants';
import { unboxAxiosResponse } from '../../../utils';

export const openJournal = ({ cashBeginningBalance, openingNote }) => {
  return axios
    .post(`${apiBasePath}/pos/terminal/journal/open`, { cashBeginningBalance, openingNote })
    .then((response) => unboxAxiosResponse(response));
};

export const closeJournal = ({ cashClosingBalance, closingNote }) => {
  return axios
    .post(`${apiBasePath}/pos/terminal/journal/close`, { cashClosingBalance, closingNote })
    .then((response) => unboxAxiosResponse(response));
};

export const getJournalSummary = () => {
  return axios.get(`${apiBasePath}/pos/terminal/journal`).then((response) => unboxAxiosResponse(response));
};
