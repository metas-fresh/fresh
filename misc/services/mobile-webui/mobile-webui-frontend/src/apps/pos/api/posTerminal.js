import axios from 'axios';
import { apiBasePath } from '../../../constants';
import { unboxAxiosResponse } from '../../../utils';

export const getPOSTerminal = () => {
  return axios.get(`${apiBasePath}/pos/terminal`).then((response) => unboxAxiosResponse(response));
};
