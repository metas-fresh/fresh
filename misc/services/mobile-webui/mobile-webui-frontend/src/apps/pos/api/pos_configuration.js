import axios from 'axios';
import { useEffect, useState } from 'react';
import { apiBasePath } from '../../../constants';
import { unboxAxiosResponse } from '../../../utils';

export const usePOSConfiguration = () => {
  const [isLoading, setLoading] = useState(false);
  const [config, setConfig] = useState([]);

  useEffect(() => {
    setLoading(true);
    getConfig()
      .then(setConfig)
      .finally(() => setLoading(false));
  }, []);

  return isLoading ? { isLoading: true } : config;
};

const getConfig = () => {
  return axios.get(`${apiBasePath}/pos/config`).then((response) => unboxAxiosResponse(response));
};
