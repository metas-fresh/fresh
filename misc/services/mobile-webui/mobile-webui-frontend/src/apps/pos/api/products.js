import axios from 'axios';
import { apiBasePath } from '../../../constants';
import { unboxAxiosResponse } from '../../../utils';
import { useEffect, useState } from 'react';
import { toUrl } from '../../../utils/index';
import { useDebounce } from '../../../hooks/useDebounce';

const queryDebounceMillis = 300;

export const useProducts = ({ onBarcodeResult }) => {
  const [isLoading, setLoading] = useState(false);
  const [queryString, setQueryStringState] = useState('');
  const [queryStringToSearch, setQueryStringToSearch] = useState('');
  const [products, setProducts] = useState({});

  const setQueryStringToSearchDebounced = useDebounce(setQueryStringToSearch, queryDebounceMillis);
  const setQueryString = (queryStringNew) => {
    setQueryStringState(queryStringNew);
    setQueryStringToSearchDebounced(queryStringNew);
  };

  useEffect(() => {
    setLoading(true);
    getProducts(queryStringToSearch)
      .then((productsNew) => {
        if (onBarcodeResult && productsNew.barcodeMatched && productsNew.list?.length === 1) {
          const product = productsNew.list[0];
          onBarcodeResult(product);
          setQueryString('');
        } else {
          setProducts(productsNew);
        }
      })
      .finally(() => setLoading(false));
  }, [queryStringToSearch]);

  return {
    isLoading,
    isBarcodeMatched: !!products?.barcodeMatched,
    list: products?.list ?? [],
    queryString,
    setQueryString: (queryStringNew) => {
      setQueryStringState(queryStringNew);
      setQueryStringToSearchDebounced(queryStringNew);
    },
  };
};

const getProducts = (query) => {
  const url = toUrl(`${apiBasePath}/pos/products`, { query });
  return axios.get(url).then((response) => unboxAxiosResponse(response));
};
