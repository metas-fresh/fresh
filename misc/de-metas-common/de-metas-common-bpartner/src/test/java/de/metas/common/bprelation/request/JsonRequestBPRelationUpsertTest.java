/*
 * #%L
 * de.metas.business.rest-api
 * %%
 * Copyright (C) 2021 metas GmbH
 * %%
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program. If not, see
 * <http://www.gnu.org/licenses/gpl-2.0.html>.
 * #L%
 */

package de.metas.common.bprelation.request;

<<<<<<< HEAD
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import de.metas.common.bprelation.JsonBPRelationRole;
import org.junit.BeforeClass;
import org.junit.Test;
=======
import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.fasterxml.jackson.databind.ObjectMapper;
import de.metas.common.bprelation.JsonBPRelationRole;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
>>>>>>> 3091b8e938a (externalSystems-Leich+Mehl can invoke a customizable postgREST reports (#19521))

import java.io.IOException;
import java.util.Collections;

<<<<<<< HEAD
import static io.github.jsonSnapshot.SnapshotMatcher.expect;
import static io.github.jsonSnapshot.SnapshotMatcher.start;
import static org.assertj.core.api.Assertions.assertThat;

public class JsonRequestBPRelationUpsertTest
{
	private final ObjectMapper mapper = new ObjectMapper();

	@BeforeClass
	public static void beforeAll()
	{
		start();
	}
=======
import static org.assertj.core.api.Assertions.assertThat;

@ExtendWith(SnapshotExtension.class)
public class JsonRequestBPRelationUpsertTest
{
	private final ObjectMapper mapper = new ObjectMapper();
	private Expect expect;
>>>>>>> 3091b8e938a (externalSystems-Leich+Mehl can invoke a customizable postgREST reports (#19521))

	@Test
	public void serializeDeserialize() throws IOException
	{
		final JsonRequestBPRelationsUpsert item = JsonRequestBPRelationsUpsert.builder()
				.orgCode("org123")
				.locationIdentifier("loc123")
				.relatesTo(Collections.singletonList(JsonRequestBPRelationTarget.builder()
						.targetBpartnerIdentifier("targetBP123")
						.targetLocationIdentifier("targetLoc123")
						.name("relName")
						.description("relDesc")
						.externalId("ext123")
						.billTo(true)
						.shipTo(true)
						.fetchedFrom(true)
						.handOverLocation(true)
						.role(JsonBPRelationRole.Caregiver)
						.payFrom(true)
						.remitTo(true)
						.active(true)
						.build()))
				.build();

		final String string = mapper.writeValueAsString(item);
		assertThat(string).isNotEmpty();

		final JsonRequestBPRelationsUpsert result = mapper.readValue(string, JsonRequestBPRelationsUpsert.class);

		assertThat(result).isEqualTo(item);
<<<<<<< HEAD
		expect(result).toMatchSnapshot();
=======
		expect.serializer("orderedJson").toMatchSnapshot(result);
>>>>>>> 3091b8e938a (externalSystems-Leich+Mehl can invoke a customizable postgREST reports (#19521))
	}
}
