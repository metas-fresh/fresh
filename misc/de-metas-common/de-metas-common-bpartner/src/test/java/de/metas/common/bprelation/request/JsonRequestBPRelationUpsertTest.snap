de.metas.common.bprelation.request.JsonRequestBPRelationUpsertTest.serializeDeserialize=[
  {
<<<<<<< HEAD
    "orgCode" : "org123",
    "locationIdentifier" : "loc123",
    "relatesTo" : [
      {
        "targetBpartnerIdentifier" : "targetBP123",
        "targetLocationIdentifier" : "targetLoc123",
        "name" : "relName",
        "description" : "relDesc",
        "externalId" : "ext123",
        "role" : "Caregiver",
        "billTo" : true,
        "fetchedFrom" : true,
        "payFrom" : true,
        "handOverLocation" : true,
        "remitTo" : true,
        "shipTo" : true,
        "active" : true
=======
    "locationIdentifier": "loc123",
    "orgCode": "org123",
    "relatesTo": [
      {
        "active": true,
        "billTo": true,
        "description": "relDesc",
        "externalId": "ext123",
        "fetchedFrom": true,
        "handOverLocation": true,
        "name": "relName",
        "payFrom": true,
        "remitTo": true,
        "role": "Caregiver",
        "shipTo": true,
        "targetBpartnerIdentifier": "targetBP123",
        "targetLocationIdentifier": "targetLoc123"
>>>>>>> 3091b8e938a (externalSystems-Leich+Mehl can invoke a customizable postgREST reports (#19521))
      }
    ]
  }
]